### PR TITLE
OSHMEM: spml ikrit: fix mxm disconnect flow

### PR DESCRIPTION
**Pull request by alex-mikheev**
@miked-mellanox @jladd-mlnx 

Add out of band barrier before performing mxm disconnect.
It will make sure that every pe is ready to disconnect. Otherwise
bad things may happen.
(cherry picked from commit ompi/open-mpi@3f7ed5654802d4b5921263daba7811f77b7751fc)

### CODE REVIEW

File: oshmem/mca/spml/ikrit/spml_ikrit.c [hunk 1]
```diff
@@ -344,6 +344,7 @@ int mca_spml_ikrit_del_procs(oshmem_proc_t** procs, size_t nprocs)
     size_t i;
     opal_list_item_t *item;
 
+    oshmem_shmem_barrier();
```
Comment (jsquyres) [inline oshmem/mca/spml/ikrit/spml_ikrit.c:347]: I don't know SPML del_procs is called, but is it guaranteed that all processes will call it at the same time?

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/76/
Test PASSed.

File: oshmem/mca/spml/ikrit/spml_ikrit.c (hunk 1 above)
Comment (alex-mikheev) [inline oshmem/mca/spml/ikrit/spml_ikrit.c:347]: Yes. It is part of shmem_finalize() which behaves like MPI_Finalize(). See "oshmem/runtime/oshmem_shmem_finalize.c"

**Comment by mike-dubman** [comment]
:+1:

**Comment by jsquyres** [comment]
Is this necessary for 1.8.4, or can it wait until 1.8.5?

(would prefer to wait, if possible)

**Comment by jladd-mlnx** [comment]
Why wait? Looks good to me.

**Pull request merged**